cmds/xinit/xinit.go

```diff
@@ -32,10 +32,21 @@ func setup() error {
 		}
 		time.Sleep(time.Second)
 	}
-	for _, f := range []string{"flwm", "opera-12"} {
+	// wingo and flwm get to fight it out. If there is no wingo, we get
+	// flwm; else we get wingo.
+	for _, f := range []string{"wingo", "flwm", "midori", "opera-12"} {
 		log.Printf("Run %v", f)
 		go func(n string) {
-			cmd := exec.Command(n, "-display", ":0")
+			// We used to set -display :0, which all X programs
+			// are supposed to accept.
+			// wingo can not take the -display option at all.
+			// It gets an error.
+			// That option doesn't work with midori.
+			// It does not get an error, it just ignores the option.
+			// It's kind of weird how many of the basic X rules
+			// are no longer known.
+			cmd := exec.Command(n)
+			cmd.Env = append(os.Environ(), "DISPLAY=:0")
```

Comment:
Do you want to remove the `-display` flag from the `aterm` command as well? It looks like absent the flag it should use the env variable. The outcome should be the same just wondering about consistency.

Reply:
let's look at this in a bit, X is so inconsistent anyway I'm not sure this matters. Putting -display in the args is marginally more convenient. The linuxcon show network is so bad I don't want to do this at present :-(

Reply:
well changing my mind, testing the change.